sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -161,140 +152,52 @@ AccessToken TokenCredentialImpl::ParseToken(
     std::string const& expiresInPropertyName,
     std::string const& expiresOnPropertyName)
 {
-  // TODO: use JSON parser.
-  AccessToken accessToken;
-  if (!GetPropertyValueAsString(jsonString, accessTokenPropertyName, accessToken.Token))
-  {
-    ThrowMissingJsonPropertyError(accessTokenPropertyName);
-  }
-
-  int64_t expiresIn = 0;
-  if (GetPropertyValueAsInt64(jsonString, expiresInPropertyName, expiresIn))
-  {
-    accessToken.ExpiresOn = std::chrono::system_clock::now() + std::chrono::seconds(expiresIn);
-    return accessToken;
-  }
-
-  if (expiresOnPropertyName.empty())
-  {
-    ThrowMissingJsonPropertyError(expiresInPropertyName);
-  }
-
-  std::string expiresOn;
-  if (!GetPropertyValueAsString(jsonString, expiresOnPropertyName, expiresOn))
+  try
   {
-    ThrowMissingJsonPropertyError(expiresInPropertyName + "\' or \'" + expiresOnPropertyName);
-  }
+    auto const parsedJson = Azure::Core::Json::_internal::json::parse(jsonString);
 
-  {
-    auto const spacePos = expiresOn.find(' ');
-    if (spacePos != std::string::npos) // LCOV_EXCL_LINE
+    if (!parsedJson.contains(accessTokenPropertyName))
     {
-      expiresOn = expiresOn.replace(spacePos, 1, 1, 'T');
+      ThrowMissingJsonPropertyError(accessTokenPropertyName);
     }
-  }
 
-  accessToken.ExpiresOn = Azure::DateTime::Parse(expiresOn, Azure::DateTime::DateFormat::Rfc3339);
-  return accessToken;
-}
-
-namespace {
-std::string::size_type GetPropertyValueStart(
-    std::string const& jsonString,
-    std::string const& propertyName);
-
-bool GetPropertyValueAsInt64(
-    std::string const& jsonString,
-    std::string const& propertyName,
-    int64_t& outValue)
-{
-  auto const valueStartPos = GetPropertyValueStart(jsonString, propertyName);
-  if (valueStartPos == std::string::npos)
-  {
-    return false;
-  }
+    AccessToken accessToken;
+    accessToken.Token = parsedJson[accessTokenPropertyName].get<std::string>();
 
-  int64_t value = 0;
-  {
-    auto const size = jsonString.size();
-    for (auto pos = valueStartPos; pos < size; ++pos)
+    if (parsedJson.contains(expiresInPropertyName))
     {
-      auto c = jsonString[pos];
-      if (c < '0' || c > '9')
-      {
-        break;
-      }
-
-      value = (value * 10) + (static_cast<int64_t>(c) - '0');
+      accessToken.ExpiresOn
+          = std::chrono::system_clock::now()
+          + std::chrono::seconds(
+                parsedJson[expiresInPropertyName].get<std::chrono::seconds::duration::rep>());
```

Comment:
This field is just an integer representing number of seconds, right? why not just use `.get<int>` or `.get<int64_t>`?

Reply:
It is implementation-defined, the only thing being said is it should be "signed integer type of at least 35 bits".
This would allow it to work correctly on all possible implementations, and to avoid any possible warnings.
https://en.cppreference.com/w/cpp/chrono/duration